src/tbp/monty/frameworks/environments/embodied_environment.py

```diff
@@ -33,6 +35,44 @@ def action_space(self):
         """Returns list of all possible actions available in the environment."""
         pass
 
+    @abc.abstractmethod
+    def add_object(
+        self,
+        name: str,
+        position: VectorXYZ = (0.0, 0.0, 0.0),
+        rotation: QuaternionWXYZ = (1.0, 0.0, 0.0, 0.0),
+        scale: VectorXYZ = (1.0, 1.0, 1.0),
+        semantic_id: Optional[str] = None,
+        enable_physics: Optional[bool] = False,
+        object_to_avoid=False,
+        primary_target_object=None,
+    ):
+        """Add an object to the environment.
+
+        Args:
+            name: The name of the object to add.
+            position: The initial absolute position of the object.
+            rotation: The initial rotation WXYZ quaternion of the object. Defaults to
+                (1,0,0,0).
+            scale: The scale of the object to add. Defaults to (1,1,1).
+            semantic_id: Optional override for the object semantic ID.
+            enable_physics: Whether to enable physics on the object. Defaults to False.
+            object_to_avoid: If True, run collision checks to ensure the object will not
+                collide with any other objects in the scene. If collision is detected,
+                the object will be moved. Defaults to False.
+            primary_target_object: If not None, the added object will be positioned to
```

Comment:
When this is passed, then the added object should *not* obscure the view of the primary target object. The understandable confusion is probably a result of the typo that exists in the following, and which would be worth also fixing:

```python
            primary_target_bb: If not None, the bounding box of the primary target
                object; passed when we're adding multiple objects, such that we ensure
                that the added object obscures the initial view of the primary target
                object (which avoiding collision alone cannot guarantee); defined by a
                list of the min and max corners
```

It should say, "that the added object does *not* obscure"

More description is provided under `find_non_colliding_positions`, bullet (ii).

Reply:
Thank you for catching this. I updated both argument descriptions.